src/stdlib_registry.cpp

```
@@ -0,0 +1,57 @@
+                                      NativeConstantEntry entry) {
+    constants_.emplace(name, entry);
+}
+
+// --- ResourceKindRegistry ---
+
+ResourceKindRegistry &ResourceKindRegistry::instance() {
+    static ResourceKindRegistry reg;
+    return reg;
+}
+
+int ResourceKindRegistry::registerKind(const char *typeName,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)